tools/autograd/templates/Functions.cpp

```diff
@@ -265,7 +265,7 @@ Tensor glu_double_backward(const Tensor & grad, const Tensor & grad_output, cons
 
 Tensor glu_double_backward_grad_output(const Tensor & grad, const Tensor & input, int64_t dim) {
   if (dim < 0) dim += input.dim();
-  auto sizes = std::vector<int64_t>{input.sizes()};
+  std::vector<int64_t> sizes = input.sizes();
```

Comment:
does `auto sizes = input.sizes()` not work?

Reply:
We want a copy because we modify it on the line below. (input.sizes() returns an IntList which is a view)

Reply:
Not sure. But they are different, because `auto` will deduce `IntList` not `std::vector<int64_t>` (because that's what ATen does), so the answer to your question depends on the code below.

Reply:
...and right below, we use `operator[]`, not supported on `IntList`. So, yes, it does matter.